MyProject/src/main/java/com/netcracker/controller/InventoryController.java

```diff
@@ -0,0 +1,265 @@
+    public Iterable<HardwareComponent> getHwComponents(){
+
+        return hwRepository.findAll();
+    }
+
+    @GetMapping("/network-elements")
+    @ApiOperation(value = "Show all network element ")
+    public Iterable<NetworkElement> getNetElements(){
+
+        return neRepository.findAll();
+    }
+
+    @GetMapping("/ne-to-hw")
+    @ApiOperation(value = "Show all ne-to-hw ")
+    public Iterable<NeToHw> getNeToHw(){
+
+        return neToHwRepository.findAll();
+    }
+
+    @GetMapping("/ne-to-interface")
+    @ApiOperation(value = "Show all ne-to-interface ")
+    public Iterable<NeToInterface> getNeToInterface(){
+
+        return neToInterRepository.findAll();
+    }
+
+    @GetMapping("/interfaces/{id}")
+    @ApiOperation(value = "Get interface by id")
+    public Interface getInterfaceById(
+            @ApiParam(value = "interface id", required = true)
+            @PathVariable String id) {
+
+        return interfaceRepository.findById(id).orElseThrow(
+                () -> new NoSuchElementException
+                        ("Unable to find interface with id: " + id));
+    }
+
+
+    @GetMapping("/hardware-components/{id}")
+    @ApiOperation(value = "Get hardware component by id")
+    public HardwareComponent getHwComponentById(
+            @ApiParam(value = "hardware component id", required = true)
+            @PathVariable String id) {
+
+        return hwRepository.findById(id).orElseThrow(
+                () -> new NoSuchElementException
+                        ("Unable to find hardware component with id: " + id));
+    }
+
+    @GetMapping("/network-elements/{id}")
+    @ApiOperation(value = "Get network element by id")
+    public NetworkElement getNetElementById(
+            @ApiParam(value = "network element id", required = true)
+            @PathVariable String id) {
+
+        return neRepository.findById(id).orElseThrow(
+                () -> new NoSuchElementException
+                        ("Unable to find network element with id: " + id));
+    }
+
+
+
+    @GetMapping("/ne-to-hw/{id}")
+    @ApiOperation(value = "Get ne-to-hw by id")
+    public NeToHw getNeToHwById(
+            @ApiParam(value = "ne-to-hw id", required = true)
+            @PathVariable String id) {
+
+        return neToHwRepository.findById(id).orElseThrow(
+                () -> new NoSuchElementException
+                        ("Unable to find ne-to-hw with id: " + id));
+    }
+
+    @GetMapping("/links/{id}")
+    @ApiOperation(value = "Get link by id")
+    public Link getLinkById(
+            @ApiParam(value = "link id", required = true)
+            @PathVariable String id) {
+
+        return linkRepository.findById(id).orElseThrow(
+                () -> new NoSuchElementException
+                        ("Unable to find link with id: " + id));
+    }
+
+    @PostMapping("/network-elements/{id}/interface")
+    @ApiOperation(value = "Add interface")
+    public Interface createInterface(
```

Comment:
Please encapsulate the interface creation logic in a separate service. Please check https://github.com/dmfe/network-inventory/blob/master/src/main/java/com/example/inventoryapp/services/impl/InventoryCRUDService.java
and use it as an example.